src/modules/sdexec/sdexec.c

```diff
@@ -755,6 +772,20 @@ static void write_cb (flux_t *h,
         flux_log (h, LOG_ERR, "sdexec.write: subprocess no longer exists");
         return;
     }
+    /* If the systemd unit has not started yet  enqueue the write request for
+     * later processing in start_continuation().
+     */
+    if (sdexec_channel_get_fd (proc->in) != -1) { // not yet claimed by systemd
```

Comment:
this could probably use a little more comment, it's not super intuitive why you'd check for ~= -1.  Perhaps mention that start_contiunation() closes the fd after acknowledgement unit has started.